File: rust_exercises/leetcode/Problem714/Solution.java
public class Solution {
    public int maxProfit(int[] prices, int fee) {
        int cash = 0;
        int hold = -prices[0];
        
        for (int i = 1; i < prices.length; i += 1) {
            cash = Math.max(cash, hold + prices[i] - fee);
            hold = Math.max(hold, cash - prices[i]);
        }
        
        return cash;
    }
}

File: rust_exercises/leetcode/Problem714/solution.rs
use std::cmp;

impl Solution {
    pub fn max_profit(prices: Vec<i32>, fee: i32) -> i32 {
        let mut cash: i32 = 0;
        let mut hold: i32 = -prices[0];
        
        for price in prices.iter().skip(1) {
            cash = cmp::max(cash, hold + *price - fee);
            hold = cmp::max(hold, cash - *price);
        }
        
        cash
    }
}
